FSW/src/util/timing.cpp

```diff
@@ -0,0 +1,237 @@
+/* timing.cpp handles the state of the payload based on timing/sensor inputs
+ * Usage:
+ *  module functionality
+ *  function definitions
+ *  put function declarations in timing.hpp
+ *  put constants/configuration declarations in config.hpp
+ * 
+ * Modules encompassed:
+ *  Science window timing
+ *
+ * Additional files needed for compilation:
+ *  config.hpp
+ *  timing.hpp
+ *  eventUtil.cpp & eventUtil.hpp
+ *  comUtil.cpp & comUtil.hpp
+ */
+
+/* - - - - - - Includes - - - - - - */
+// All libraries are put in timing.hpp
+// NS2 headers
+#include "../headers/config.hpp"
+#include "../headers/timing.hpp"
+
+/* Module Variable Definitions */
+
+/* - - - - - - Class Implementations - - - - - - */
+/* - - - - - - ScienceMode - - - - - - *
+ * Usage:
+ *  tracks the Science Mode of the payload
+ */
+ScienceMode::ScienceMode(){ // constructor
+    mode = STANDBY_MODE;
+}
+
+int ScienceMode::get(){
+    return mode;
+}
+
+void ScienceMode::set(int newMode){
+    mode = newMode;
+}
+
+/* - - - - - - Helper Functions - - - - - - */
+
+/* - - - - - - updatePayloadMode - - - - - - *
+ * Usage:
+ *  returns the new mode that the payload should be in given the current mode and the buffer 
+ *  invokes/starts any events that are used in transitioning payload mode 
+ *  acts on scienceMode declared in config.hpp
+ * 
+ * Inputs:
+ *  buffer - buffer array containing most recent photodiode voltage measurements
+ *  bufIdx - index of most recent measurement updated in bufIdx
+ * 
+ * Outputs:
+ *  none
+ */
+void updatePayloadMode(float buffer[], int bufIdx){
+
+    int currentMode = scienceMode.get();
+    
+    if ((currentMode < MODE_NOT_RECOGNIZED) && (currentMode > STANDBY_MODE)){
+        
+        /*FUTURE TEAMS: Flesh out the below if statement with the logic for 
+        *   controlling the payload mode based on ADCS outputs 
+        * For ground testing this is hardcoded as true in config.hpp,
+        *   but once ADCS is implemented you will need to dynamically update this,
+        *   i.e.    ADCS_READY = (digitalRead(BUS_READY_PIN) == HIGH); */
+        if (ADCS_READY_FOR_SCIENCE){

```

Comment:
Variable name is maybe a bit confusing. Is this ADCS as in attitude determination, or ADC's as in multiple analog to digital converters. I don't know if I get the concept of the ADCS being ready for science? I guess the ADCS has to trigger bus reaction wheels to start scanning at the same time science is being collected, so in that case it's valid.

Reply:
made it more clear as "ADCS_POINTING_AT_SUN", in general in this code ADCS is always attitude determination and control system. I will write some proper documentation on how the ADCS should be implemented with this existing code so that future teams can hit the ground running